pipewire/qubes-pw-module.c

```diff
@@ -303,8 +332,40 @@ struct impl {
 #endif
 
     atomic_uint_fast64_t reference_count;
+
+    const char *args;
 };
 
+int get_domid_from_props(struct pw_properties *props) {
+    size_t domid;
+    const char *peer_domain_prop = NULL;
+    if ((peer_domain_prop = pw_properties_get(props, QUBES_AUDIOVM_PW_KEY)) == NULL) {
+        return -EINVAL;
+    }
+    if (parse_number(peer_domain_prop, INT_MAX / 2, &domid, "domain ID")) {
+        pw_log_debug("Cannot parse domid");
+        return -errno;
+    }
+    return domid;
+}
+
+int get_domid_from_qdb(qdb_handle_t qdb) {
+    size_t domid;
+
+    char *qdb_entry = qdb_read(qdb, QUBES_AUDIOVM_QUBESDB_ENTRY, NULL);
+    if (qdb_entry != NULL) {
+        if (parse_number(qdb_entry, INT_MAX / 2, &domid, "domain ID")) {
+            pw_log_debug("Cannot parse domid");
```

Comment:
This should have `error` severity, as it means something is quite wrong.  Please log the entry, too.